arrows/serialize/json/register_algorithms.cxx

```diff
@@ -37,43 +39,45 @@ KWIVER_SERIALIZE_JSON_PLUGIN_EXPORT
 void
 register_factories( kwiver::vital::plugin_loader& vpm )
 {
-  kwiver::vital::serializer_registrar reg( vpm, "arrows.serialize.json",
-                                           "json" );
+  auto const module_name = std::string{ "arrows.serialize.json" };
+  kwiver::vital::serializer_registrar sreg( vpm, module_name, "json" );
+  kwiver::vital::algorithm_registrar areg( vpm, module_name );
 
-  if (reg.is_module_loaded())
+  if( sreg.is_module_loaded() )
```

Comment:
FYI for other reviewers; yes, it's fine that we only use `sreg` to deal with is-loaded / mark-loaded. These just pass the module name to relevant methods of the plugin manager, and the module name is the same for both `sreg` and `areg`, so we don't need to check/use both, and `sreg`/`areg` may be used interchangeably for these operations.

Reply:
I was not aware that we had a separate registration for serialization functions.  Independent of registering the algorithm `metadata_map_io` should we also register the serialization functions for metadata types with `sreg`?  I haven't looked enough at how this serialization stuff works, but it seems like we might want something like
```c++
sreg.register_algorithm< metadata >();
sreg.register_algorithm< metadata_map >();
```
to allow people to serialize those types outside of using the `metadata_map_io` algorithm.  Is that right, or am I misunderstanding?

Reply:
I think we want to register `metadata` but `metadata_map` is not actually a class. The functionality to serialize a metadata map is contained in `metadata`. It's possible to register `metadata` and have this build with passing tests, but I'm not sure that means that it's actually working as expected. I'm a bit surprised that it wasn't registered before, because that class preceded this PR. My only concern is that `metadata` provides additional methods, (`serialize_meta`, `serialize_map`, and `deserialize_map`) on top of the `serialize` and `deserialize` provided by the other classes. My guess is this won't break anything but I think it's likely that this will only be able to serialize a vector of metadata items, e.g. the information from one metadata packet, since this is what `serialize`/`deserialize` handles.

Reply:
okay, it's not important for this PR (which I've already merged).  I was just wondering if that was something we overlooked for general JSON serialization purposed.  I don't know of enough about what registering serialization algorithms actually does in KWIVER.  It may make sense to register `metadata` if that's just a one line addition and doesn't break any tests.  Or we could just wait until someone finds a need for this.